control.lua

```diff
@@ -133,6 +140,34 @@ function get_selected_ent(pindex, ent_no)
    end
 end
 
+--- Produce an iterator over all valid entities for a player's selected tile,
+--  filtering out the player themselves.
+local function iterate_selected_ents(pindex)
+   local tile = players[pindex].tile
+   local ents = tile.ents
+   local i = 1
+
+   local next_fn
+   next_fn = function()
+      -- Ignore all entities that are a character belonging to this player. It
+      -- should only be one, but the loop doesn't hurt.
+      for j = i, #ents do
+         i = j
+
+         if not ents[i].valid then goto continue end
+
+         if ents[i].type ~= "character" or ents[i].player ~= pindex then return ents[i] end
+
+         ::continue::
+      end
+
+      return nil
+   end
+
+   if not next(ents) then return next_fn, nil, nil end
+   return next_fn, nil, nil
```

Comment:
Why return nil, nil? Is it how the for loop works when you call an iterator like this?